src/ast/dep.rs

```diff
@@ -241,10 +243,15 @@ impl Visit for DependencyCollector<'_> {
   fn visit_call_expr(&mut self, node: &ast::CallExpr) {
     node.visit_children_with(self);
 
-    let is_esm = matches!(&node.callee, Callee::Import(_));
-    if !is_esm && !self.is_require(&node.callee) {
-      return;
-    }
+    let kind = match &node.callee {
+      Callee::Import(import) => match import.phase {
+        ImportPhase::Evaluation => DynamicDependencyKind::Import,
+        ImportPhase::Source => DynamicDependencyKind::ImportSource,
+        ImportPhase::Defer => return,
```

Comment:
We might want to pre-emptively support import defer here and just have it fail at runtime.